Editor/Completion/XmlCompletionCommitManager.cs

```diff
@@ -112,5 +141,28 @@ void InsertClosingTags (IAsyncCompletionSession session, ITextBuffer buffer, Com
 			bufferEdit.Replace (span, sb.ToString ());
 			bufferEdit.Apply ();
 		}
+
+		private static void ShiftCaret (IAsyncCompletionSession session, int len, XmlCaretDirection caretDirection)
+		{
+			switch (caretDirection) {
+			case XmlCaretDirection.Left:
+				for (int i = 0; i < len; i++) {
+					session.TextView.Caret.MoveToPreviousCaretPosition ();
+				}
+				return;
+			case XmlCaretDirection.Right:
+				for (int i = 0; i < len; i++) {
+					session.TextView.Caret.MoveToNextCaretPosition ();
+				}
+				return;
+			case XmlCaretDirection.Top:
+				//TO DO
```

Comment:
I would suggest throwing `ArgumentException ($"Unsupported value '{caretDirection}'")` for unhandled value and removing the `Top` and `Down` values from the `XmlCaretDirection` enum - they can be added if they're needed

Reply:
I have included `throw new ArgumentException ($"Unsupported value '{caretDirection}'")`, although, I am letting `top` and `down` in the enum values as I get an error in `case CaretDirection.Top:` or `case CaretDirection.Down:`

@mhutch, If I need to remove enum values, I also need to remove the above case statements. Let me know what you prefere?

Reply:
I'd suggest removing them, since we currently have neither a need for them nor an implementation.

Reply:
Okay. I removed them